application/backend/tests/unit/endpoints/test_datasets.py

```diff
@@ -95,26 +95,30 @@ def test_list_dataset_items_filtering_and_pagination(self, fxt_dataset_item, fxt
         fxt_dataset_service.list_dataset_items.return_value = [fxt_dataset_item]
 
         response = fxt_client.get(
-            f"/api/projects/{str(project_id)}/dataset/items?limit=50&offset=2&start_date=2025-01-09T00:00:00Z&end_date=2025-12-31T23:59:59Z"
+            f"/api/projects/{str(project_id)}/dataset/items?limit=50&offset=2&start_date=2025-01-09T00:00:00Z&end_date=2025-12-31T23:59:59Z&annotation_status=reviewed"
         )
 
         assert response.status_code == status.HTTP_200_OK
         fxt_dataset_service.count_dataset_items.assert_called_once_with(
             project_id=project_id,
             start_date=datetime(2025, 1, 9, 0, 0, 0, tzinfo=ZoneInfo("UTC")),
             end_date=datetime(2025, 12, 31, 23, 59, 59, tzinfo=ZoneInfo("UTC")),
+            annotation_status='reviewed',
         )
         fxt_dataset_service.list_dataset_items.assert_called_once_with(
             project_id=project_id,
             limit=50,
             offset=2,
             start_date=datetime(2025, 1, 9, 0, 0, 0, tzinfo=ZoneInfo("UTC")),
             end_date=datetime(2025, 12, 31, 23, 59, 59, tzinfo=ZoneInfo("UTC")),
+            annotation_status='reviewed',
         )
 
     @pytest.mark.parametrize("limit", [1000, 0, -20])
     def test_list_dataset_items_wrong_limit(self, fxt_dataset_service, fxt_client, limit):
-        response = fxt_client.get(f"/api/projects/{uuid4()}/dataset/items?limit=${limit}")
+        response = fxt_client.get(f"/api/projects/{uuid4()}/dataset/items?limit={limit}&annotation_status=None")
+        assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
+        fxt_dataset_service.list_dataset_items.assert_not_called()
```

Comment:
This change looks weird: `annotation_status=None` is not a valid parameter value, and this test (`test_list_dataset_items_wrong_limit`) specifically focuses on the `limit` parameter.

Unit tests should be small and focus on individual pieces of functionalities. If you're not familiar with software testing, [here](https://martinfowler.com/testing/) is a highly recommended read (also recommended for seasoned developers, to be fair). 

Please write a dedicated unit test for this parameter; if you're fully relying on an AI model to write the code, this is a great moment to step in as a human developer :)